yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/table-index-sizes.sqlplus

```diff
@@ -13,18 +13,26 @@ WHENEVER OSERROR EXIT FAILURE
 
 -- Select the data into the CSV file using inline views
 SPOOL table-index-sizes.csv
+
+-- Inline view to get the base tables, partitions, and subpartitions with correct naming
 SELECT schema_name, object_name, object_type, size_in_bytes
 FROM (
     SELECT
-        owner AS schema_name,
-        segment_name AS object_name,
-        'table' AS object_type,
-        bytes AS size_in_bytes
+        s.owner AS schema_name,
+        CASE
+            WHEN s.segment_type = 'TABLE' THEN s.segment_name
+            WHEN s.segment_type = 'TABLE PARTITION' THEN s.segment_name || '_' || s.partition_name
+            WHEN s.segment_type = 'TABLE SUBPARTITION' THEN s.segment_name || '_' || sp.partition_name || '_' || s.partition_name
+        END AS object_name,
+        s.segment_type AS object_type,
+        s.bytes AS size_in_bytes
     FROM
-        dba_segments
+        dba_segments s
+    LEFT JOIN
+        dba_tab_subpartitions sp ON s.owner = sp.table_owner AND s.segment_name = sp.table_name AND s.partition_name = sp.subpartition_name
     WHERE
-        segment_type IN ('TABLE', 'TABLE PARTITION', 'TABLE SUBPARTITION')
-        AND owner = '&1'
+        s.segment_type IN ('TABLE', 'TABLE PARTITION', 'TABLE SUBPARTITION')
```

Comment:
I don't understand what "INDEX PARTITION" really means, but should that also be included in this list?

Reply:
added one more commit, no need of this type now.